xgitguard/custom keyword search/public_keyword_search.py

```diff
@@ -0,0 +1,442 @@
+    """
+    Run GitHub search
+    If a primary keyword is provided, perform the search using the primary keyword.
+    params: public_keywords - list - optional
+    params: org - list - optional
+    params: repo - list - optional
+    returns: True or False
+    """
+    if public_keywords:
+        if isinstance(public_keywords, list):
+            configs.secondary_keywords = public_keywords
+        else:
+            logger.error(f"Please pass public_keywords in List like '['password',]'")
+            sys.exit(1)
+    else:
+        # Get the secondary_keywords from secondary_keywords file
+        configs.read_secondary_keywords(file_name="public_keywords.csv")
+    logger.info(f"Total Public keywords : {len(configs.secondary_keywords)}")
+
+    total_search_pairs = len(configs.secondary_keywords)
+    logger.info(f"Total Search Pairs: {total_search_pairs}")
+
+    total_processed_search, total_detection_writes = 0, 0
+    search_query_list = []
+    # Format GitHub Search Query List
+    search_query_list = format_search_query_list(configs.secondary_keywords)
+    logger.info(f"Total No.of search queries: {len(search_query_list)}")
+
+    # Loop over each search query
+    for search_query in search_query_list:
+        detection_writes_per_query = 0
+        new_results_per_query = 0
+        logger.info(f"*******  Processing Search Query: {search_query}   *******")
+        try:
+            # Search GitHub and return search response confidence_score
+            total_processed_search += 1
+            time.sleep(2)
+            search_response_lines = githubCalls.run_github_search(
+                search_query,
+                "",
+                org,
+                repo,
+            )
+            # If search has detections, process the result urls else continue next search
+            if search_response_lines:
+                (
+                    detection_writes_per_query,
+                    new_results_per_query,
+                    detections_per_query,
+                ) = process_search_results(
+                    search_response_lines,
+                    search_query,
+                )
+                logger.info(
+                    f"Detection writes in current search query: {detection_writes_per_query}"
+                )
+                total_detection_writes += detection_writes_per_query
+            else:
+                logger.info(
+                    f"Search '{search_query}' returns no results. Continuing..."
+                )
+                continue
+        except Exception as e:
+            logger.error(f"Process Error: {e}")
+        if new_results_per_query >= 0:
+            logger.info(
+                f"Total: {total_search_pairs} "
+                + f"Processed: {total_processed_search} "
+            )
+    logger.info(f"Current Total Processed Search: {total_processed_search}")
+    logger.info(f"Current Total Detections Write: {total_detection_writes}")
+
+    return True
```

Comment:
Please keep the changes similar to enterprise keyword search module.

Reply:
updated